core/src/main/java/org/sonarsource/sonarlint/core/SonarLintBackendImpl.java

```diff
@@ -105,6 +110,8 @@ public SonarLintBackendImpl(SonarLintClient client) {
     clientEventBus.register(rulesRepository);
     clientEventBus.register(sonarProjectBranchService);
     clientEventBus.register(synchronizationServiceImpl);
+    this.storageFacade = new StorageFacade();
+    this.analysisService = new AnalysisServiceImpl(configurationRepository, storageFacade);
```

Comment:
Nitpick: maybe move these initialization statements before the registrations to the event bus?